turbine/src/cluster_nodes.rs

```diff
@@ -661,13 +735,52 @@ mod tests {
             vec![24, 32],
             vec![34],
         ];
-        for (k, peers) in peers.into_iter().enumerate() {
-            let retransmit_peers = get_retransmit_peers(/*fanout:*/ 3, k, &index);
-            assert_eq!(retransmit_peers.collect::<Vec<_>>(), peers);
-        }
-        for k in 13..=index.len() {
-            let mut retransmit_peers = get_retransmit_peers(/*fanout:*/ 3, k, &index);
-            assert_eq!(retransmit_peers.next(), None);
-        }
+        check_retransmit_nodes(/*fanout:*/ 3, &nodes, peers);
+        let nodes = [
+            5, // root
+            34, 52, 8, // 1st layer
+            // 2nd layar
+            44, 18, 2, // 1st neigborhood
+            42, 47, 46, // 2nd
+            11, 26, 28, // 3rd
+            // 3rd layer
+            53, 23, 37, // 1st neighborhood
+            40, 13, 7, // 2nd
+            50, 35, 22, // 3rd
+            3, 27, 31, // 4th
+            10, 48, 15, // 5th
+            19, 6, 30, // 6th
+            36, 45, 1, // 7th
+            38, 12, 17, // 8th
+            4, 32, 16, // 9th
+            // 4th layer
+            41, 49, 24, // 1st neighborhood
+            14, 9, 0, // 2nd
+            29, 21, 39, // 3rd
+            43, 51, 33, // 4th
+            25, 20, // 5th
+        ];
+        let peers = vec![
```

Comment:
hard coding the expected children is painful.
Seems like we should be able to do something like:
```
peers = get_retransmit_peers(y)
for peer in peers
    assert!(y == get_retransmit_parent(peer))
```
This could allow sweeping across different fanout and node numbers without a crazy amount of hard coding.

Reply:
That will only verify that `get_retransmit_peers` and `get_retransmit_parent` are inverse of each other.
We still need to verify that at least one of them is correct, which requires hard-coding either the parents are the children.

I added a "round-trip" test to verify `get_retransmit_peers` and `get_retranmsit_parent` are inverse of each other.

Reply:
Yeah, makes sense. Hard coding to make sure get_retransmit_peers creates the actual turbine tree we expect. Then round-trip test to verify the inverse function property.

I think the value here is that we can verify a large number of cases for the round-trip test with minimal effort (no hard coding)